mongodb/mongodb.rb

```diff
@@ -119,97 +111,119 @@ def parent_interrupt
 # Look for config file
 @config = YAML.load(File.open(config_file))
 
-if !@config.nil?
+# Update config to add mongodb_admin service
+if @config.key? 'mongodb'
+  @config['mongodb_admin'] = {}
+
+  @config['mongodb_admin']['group_name'] = @config['mongodb']['group_name'] + '_admin'
+  @config['mongodb_admin']['group_label'] = @config['mongodb']['group_label'] + ' Admin'
+  @config['mongodb_admin']['dashboard'] = @config['mongodb']['dashboard'] + ' Admin'
+  @config['mongodb_admin']['servers'] = []
+
+  @config['mongodb']['servers'].each do |server|
+    single_server = {}
+    single_server['name'] = server['name']+'-admin'
+    single_server['hostname'] = server['hostname']
+    single_server['port'] = server['port']
+
+    single_server['username'] = server['databases'][0]['username']
+    single_server['password'] = server['databases'][0]['password']
+
+    @config['mongodb_admin']['servers'].push single_server
+  end
+
+end
+
+unless @config.nil?
```

Comment:
This if block should be before the previous one. If there is no copperegg entry, there is no point in processing ahead.